jobserver/models.py

```diff
@@ -145,5 +183,110 @@ class JobOutput(models.Model):
     )
 
 
+class JobRequest(models.Model):
+    """
+    A request to run a Job
+
+    This represents the request, from a Human, to run a given Job in a
+    Workspace.  The job-runner will create any required Jobs for the requested
+    one to run.  All Jobs, either added by Human or Computer, are then grouped
+    by this object.
+    """
+
+    created_by = models.ForeignKey(
+        "User",
+        null=True,
+        blank=True,
+        on_delete=models.CASCADE,
+    )
+    workspace = models.ForeignKey(
+        "Workspace", on_delete=models.CASCADE, related_name="job_requests"
+    )
+
+    created_at = models.DateTimeField(auto_now_add=True)
+
+    @property
+    def completed_at(self):
+        return first(reversed(self.ordered_jobs)).completed_at
+
+    @property
+    def is_complete(self):
+        return all(j.is_complete for j in self.jobs.all())
+
+    @property
+    def is_failed(self):
+        return any(j.is_failed for j in self.jobs.all())
```

Comment:
This could be confusing as it is possible for a group to have failed items while it is still in progress, but "failed" implies "completed".

We do want action graphs to continue on failure where possible, so I think the fix is for this property to be called something like `has_any_failures`, and the state shown the user to distinguish between "in progress (no failures)" and "in progress (with failures)"

Reply:
Fixed and rebased 76bf7e0.  I've kept it simple for now by only defining failed as "all jobs have completed with 1 or more failures".  I'll revisit this after this is merged to add the "in progress with failures" state.